common/src/sql/dbinit.sql

```diff
@@ -463,15 +457,17 @@ CREATE TABLE omicron.public.Region (
 /*
  * Allow all regions belonging to a disk to be accessed quickly.
  */
-CREATE INDEX on omicron.public.Region (
-    volume_id
+CREATE UNIQUE INDEX on omicron.public.region (
+    volume_id,
+    id
```

Comment:
This is one of the cases mentioned in the PR description.  To paginate the regions on a disk, the id needs to be here, and then this can be a `UNIQUE` index.